ament_cmake_version/cmake/ament_cmake_version.cmake

```diff
@@ -0,0 +1,107 @@
+################################################################################
+# `ament_cmake_version` function creates and installs a version header file.
+# The version is taken from `package.xml` file's `<version>` tag
+# The function uses an existing "resource/version.h.in" template file to generate
+# the destination `version.h` file and installs that generated file into `DESTINATION include`.
+# The generated file is being (re-)created if:
+# - the file does not exist
+# - the file does exists but contains a version that differs from the version in `package.xml` file
+################################################################################
+# Example of a template file for `rcutils` project:
+# // Copyright 2015 Open Source Robotics Foundation, Inc.
+# //
+# // Licensed under the Apache License, Version 2.0 (the "License");
+# // you may not use this file except in compliance with the License.
+# // You may obtain a copy of the License at
+# //
+# //     http://www.apache.org/licenses/LICENSE-2.0
+# //
+# // Unless required by applicable law or agreed to in writing, software
+# // distributed under the License is distributed on an "AS IS" BASIS,
+# // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# // See the License for the specific language governing permissions and
+# // limitations under the License.
+#
+# #ifndef RCUTILS__VERSION_H_
+# #define RCUTILS__VERSION_H_
+#
+# /// \def RCUTILS_VERSION_MAJOR
+# /// Defines RCUTILS major version number
+# #define RCUTILS_VERSION_MAJOR (@rcutils_VERSION_MAJOR@)
+#
+# /// \def RCUTILS_VERSION_MINOR
+# /// Defines RCUTILS minor version number
+# #define RCUTILS_VERSION_MINOR (@rcutils_VERSION_MINOR@)
+#
+# /// \def RCUTILS_VERSION_PATCH
+# /// Defines RCUTILS version patch number
+# #define RCUTILS_VERSION_PATCH (@rcutils_VERSION_PATCH@)
+#
+# /// \def RCUTILS_VERSION_STR
+# /// Defines RCUTILS version string
+# #define RCUTILS_VERSION_STR "@rcutils_VERSION@"
+#
+# #endif  // RCUTILS__VERSION_H_"
+
+function(ament_cmake_version)
```

Comment:
It would be good to explicitly check that no arguments have accidentally passed to this function.